### ISSUE TITLE
Apple Pay - Modal Displays Incorrect ISK Amounts

### ISSUE THREAD

**Issue by dimitri-dev** [Adyen/adyen-web#2922]
**Describe the bug**
Since Adyen Web version 6.1.0, the Apple Pay modal displays incorrect amounts for transactions in Icelandic Króna (ISK) currency. This issue arises due to the handling of minor units. A recent PR (#2811) defines ISK as a ZERO_DECIMAL_CURRENCY. As a result, the Apple Pay modal shows an inflated amount to the customer. @jureperak pointed this out as well.

Everywhere in Adyen, ISK is used with two (2) decimals in minor amount, which is different from the ISO specification. Please reference: [Adyen Currency Codes - Documentation](https://docs.adyen.com/development-resources/currency-codes/)
```
ISK | Iceland Krona | 2 (differs from standard)
```

The reason why this issue occurs due to the PR is pretty much summed up in these lines:
https://github.com/Adyen/adyen-web/blob/4fb7df49ba2fc0a51cc8766a05b78f9a6a799647/packages/lib/src/components/ApplePay/payment-request.ts#L4-L8C29

The script formats the ISK amount for the apple pay modal using the changed (now: 0) decimals rule.

**To Reproduce**
1. Implement a website with the Adyen Drop-In component.
2. Initiate a transaction using ISK as the currency.
3. Set the `amount.value` as the payment value in minor units (e.g., 10,000 for a 100 ISK transaction).
4. Observe the displayed amount in the Apple Pay modal.

**Expected behavior**
The Apple Pay modal should correctly interpret the minor units for ISK, displaying 100 ISK for a 10,000 minor unit input, aligning with the two-decimal convention rather than interpreting ISK as a zero-decimal currency.

**Desktop (please complete the following information):**
- Tested on MacOS (Macbook Pro M1) in Safari browser.

**Smartphone (please complete the following information):**
- Tested on iPhone 12 Pro running on iOS 17.6.1, in Safari browser.

**Additional context**
This issue should be very easy to reproduce by configuring a test payment in ISK.
I think it's clear why showing 10,000 ISK instead of 100 ISK to customers trying to make a purchase is a show-stopper.
**The customer is properly charged (100 ISK) in the end**, but seeing a 100x larger amount will probably cause him to abandon payment.

**Comment by sponglord** [comment]
@dimitri-dev @jureperak 
Thanks for your feedback. We are reverting this change in the next release

**Comment by dimitri-dev** [comment]
@sponglord Hey there. Have you tested this by actually completing a transaction in ISK?

We have reverted to v6.0.3 (before PR #2811 was merged) and the modal used to display the amount correctly.
Since your PR #2924 reverts those changes, I assume that part is fine.

However, what bothers @jureperak and me at the moment, is that we can't get a transaction on v6.0.3 processed.
While the modal is shown correctly, upon sending a `/payment` request to Adyen, we get status `422`, errorCode `5_001` and a message stating `ApplePay token amount-mismatch`.

Since ApplePay component deals with decimal amount, and we send in minor unit requests to Adyen backend, our current assumption is that the backend isn't checking the amounts properly (comparing Apple token's decimal amount to our minor units amount). Perhaps they don't have ISK configured properly on that end?
